cargo-miri/bin.rs

```diff
@@ -452,6 +472,43 @@ path = "lib.rs"
     }
 }
 
+/// Detect the target directory by calling `cargo metadata`.
+fn detect_target_dir() -> PathBuf {
+    #[derive(Deserialize)]
+    struct Metadata {
+        target_directory: PathBuf,
+    }
+    let mut cmd = cargo();
+    // `-Zunstable-options` is required by `--config`.
+    cmd.args(["metadata", "--no-deps", "--format-version=1", "-Zunstable-options"]);
```

Comment:
I realized that the (now removed as part of 34217bdc8e088051e831de7baa33c377724af3e0) "we need a nightly Cargo to run tests that depend on unstable Cargo features" comment was not accurate. This is passing `-Zunstable-options` without checking whether `--config` is passed, so nightly is always required (but I think it's easy to add an `if` for this, although there is still one test using the `--config` flag, so nightly is still required by the test). I somehow forgot this fact when adding that comment, so I'm clarifying here.

Reply:
Yeah but Miri as a nightly-only tool requiring nightly-only cargo is okay... in the past this made `cargo tree` unusable in the Miri repo, but now that that is integrated into cargo I am less worried about such effects.

If an issue comes up we can always try to improve compatibility with stable cargo.